client/tokenized-payment-request/cart-api.js

```diff
@@ -79,10 +80,17 @@ export default class PaymentRequestCartApi {
 	 * @return {Promise} Cart response object.
 	 */
 	async getCart() {
-		return await this._request( {
+		const response = await this._request( {
 			method: 'GET',
 			path: '/wc/store/v1/cart',
+			parse: false,
 		} );
+
+		this.cartRequestHeaders = {
+			Nonce: response.headers.get( 'Nonce' ),
```

Comment:
What was the reason for copying the `Nonce` header, in this case?

The `Nonce` field retrieved from the request should also be copied in the `X-WooPayments-Store-Api-Nonce` property of `cartRequestHeaders` - similar to this: https://github.com/Automattic/woocommerce-payments/blob/b81abb4a3426c2bd47cc9a3cf5533e53cd7f5287/client/tokenized-payment-request/cart-api.js#L107-L107

Reply:
`Nonce` was copied here because it's needed for [the `updateCustomer()` call](https://github.com/Automattic/woocommerce-payments/blob/b81abb4a3426c2bd47cc9a3cf5533e53cd7f5287/client/tokenized-payment-request/payment-request.js#L235) that happens after clicking the the PRB. Without the nonce, that request fails with a nonce error. 

I see that `X-WooPayments-Store-Api-Nonce` will also work, since it overrides the `Nonce` when present, but I'm not sure it's needed with `getCart()` as `Nonce` alone works fine in block based themes. @frosso Can you refresh my memory on which point in the PRB flow the nonce error was occurring with block based themes that prompted the need to override it? If it's happening on the cart page as well, we can just use `X-WooPayments-Store-Api-Nonce` unless you think we should just use it regardless?

>AFAICT, the only way I can see this occurring is if we have cachedCartData and the nonce has been refreshed. Is this a feasible scenario or did you discover any other circumstance on the shortcode checkout that required us to fetch the nonce anew, @mdmoore?

On the cart page, a nonce has not yet been set in `cartRequestHeaders` by any previous request. The PDP uses `createAnonymousCart()` to add the nonce values to `cartRequestHeaders` but the cart page doesn't use the anonymous cart request and instead uses `getCart()`. So in short, whether adding a product to the cart on the shop page or from within the PDP, the nonce still must be copied to the headers so that it's present for subsequent Store API requests.

Reply:
I think I understand the problem a bit better.
The thing with the `Nonce` is that it's needed for `POST`/`PUT`/`DELETE` requests. It's not needed for `GET` requests.
There is a scenario where the `Nonce` is automatically saved, added, and updated to each request to the Store API: when [this middleware](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-blocks/assets/js/middleware/store-api-nonce.js) is present on the page.
The middleware is always present when a block-based theme is used, or when a cart or checkout block is present on the page.
Since this implementation is for the classic checkout, the middleware is not present on the page. 
[This PR](https://github.com/Automattic/woocommerce-payments/pull/8840) has a bit more background on this issue.

Using the `Nonce` from the `getCart` response seems to be a good solution to me 👍 

It's unfortunate that in this scenario, the `Nonce` is not automagically provided by the middleware (because the middleware is not loaded).

Reply:
Ah, that makes sense as to why I wasn't seeing the nonce added automatically - no magic on the shortcode pages. Thanks for the explanation!